### PR TITLE
Histogram & gauge metrics

### PR DESCRIPTION
**Pull request by tomahawk28**
### How to test

1. Run the operator
```
kubeconfig=`cat $(kind get kubeconfig-path --name="kind")` && go run main.go daemon --service.circleci.auth.token=xyz | luigi
```
2. Run prometheus
```
cat prom.yml
scrape_configs:
  - job_name:       'example-random'

    static_configs:
      - targets: ['localhost:8000']
        labels:
          group: 'production'

./prometheus --config.file=prom.yml
```

3. Check it works by connecting http://localhost:9090/graph

### CODE REVIEW

File: service/collector/build.go [hunk 1]
```diff
@@ -3,40 +3,121 @@ package collector
...
+		},
+		nil,
+	)
+)
+
+var (
+	buildDesc = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    prometheus.BuildFQName(namespace, subsystem, "buildtime"),
+			Help:    "circle CI build time duration.",
```
Comment (tomahawk28) [inline service/collector/build.go:27]: Not sure what's the best description would be in here.

File: service/collector/build.go [hunk 2]
```diff
@@ -3,40 +3,121 @@ package collector
...
+		logger:         config.Logger,
 	}
 
 	return c, nil
 }
 
 // Collect is the main metrics collection function.
 func (c *Build) Collect(ch chan<- prometheus.Metric) error {
-	// TODO
+	builds, err := c.circleCIClient.ListRecentBuilds(100, 0)
```
Comment (pipo02mix) [inline service/collector/build.go]: What about parameterizing these arguments as constants? It easier to tune later

Comment (tomahawk28) [reply service/collector/build.go]: Actually, this 100 is their maximum number we can request, but parameterizing makes sense. 😆

File: service/collector/build.go [hunk 3]
```diff
@@ -3,40 +3,121 @@ package collector
 import (
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
+	"github.com/jszwedko/go-circleci"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var (
+	buildLabelDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystem, "build"),
```
Comment (JosephSalisbury) [inline service/collector/build.go]: `build_info` would be nicer

File: service/collector/build.go [hunk 4]
```diff
@@ -3,40 +3,121 @@ package collector
...
+			labelState,
+		},
+		nil,
+	)
+)
+
+var (
+	buildDesc = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    prometheus.BuildFQName(namespace, subsystem, "buildtime"),
```
Comment (JosephSalisbury) [inline service/collector/build.go]: `build_time_seconds` would be better, to follow prometheus guidelines

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**